src/main/java/com/google/neighborgood/servlets/MessageServlet.java

```diff
@@ -106,4 +107,38 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     datastore.put(msgEntity);
     response.sendRedirect(request.getHeader("Referer"));
   }
+
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    // First check whether the user is logged in
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect(userService.createLoginURL("/account.jsp"));
+      return;
+    }
+
+    // Get the task ID
+    String taskId = request.getParameter("task-id");
+    if (taskId == null) {
+      System.err.println("The task id is not included");
+      return;
+    }
+
+    // Get all stored entities related with the task that corresponds to the given taskId
+    Filter filter = new FilterPredicate("taskId", FilterOperator.EQUAL, taskId);
+    Query query = new Query("Message").setFilter(filter);
```

Comment:
I don't know how much of a difference this could make, but Datastore has a special type of query to only returns entity's keys which is supposed to be lower-latency, so if you're thinking about scaling issues it might be worth looking into this as well: https://cloud.google.com/appengine/docs/standard/java/datastore/queries#keys-only_queries

Reply:
Cool! I will try to use that in the new commit.